### PR TITLE
Use ReflectTypeDescriptor in TypeDescriptor

### PR DESCRIPTION
**Pull request by twsouthwick**
@chlowell

### CODE REVIEW

File: src/System.ComponentModel.TypeConverter/src/System.ComponentModel.TypeConverter.csproj [hunk 1]
```diff
@@ -72,6 +72,10 @@
     <Compile Include="System\ComponentModel\ReflectTypeDescriptionProvider.cs" />
     <Compile Include="System\ComponentModel\RefreshEventArgs.cs" />
     <Compile Include="System\ComponentModel\RefreshEventHandler.cs" />
+    <Compile Include="System\ComponentModel\ReflectTypeDescriptionProvider.ReflectedTypeData.cs" />
+    <Compile Include="System\ComponentModel\ReflectPropertyDescriptor.cs" />
+    <!--<Compile Include="System\ComponentModel\RefreshEventArgs.cs" />-->
+    <!--<Compile Include="System\ComponentModel\RefreshEventHandler.cs" />-->
```
Comment (chlowell) [inline src/System.ComponentModel.TypeConverter/src/System.ComponentModel.TypeConverter.csproj]: These commented includes are redundant.

File: src/System.ComponentModel.TypeConverter/src/Resources/Strings.resx [hunk 2]
```diff
@@ -156,6 +165,18 @@
...
+  </data>
+  <data name="TypeDescriptorUnsupportedRemoteObject" xml:space="preserve">
+    <value>The object {0} is being remoted by a proxy that does not support interface discovery.  This type of remoted object is not supported.</value>
+  </data>
+  <data name="InvalidNullArgument" xml:space="preserve">
+   <value>Null is not a valid value for {0}.</value> 
+  </data>
+  <data name="MetaExtenderName" xml:space="preserve">
+   <value>{0} on {1}</value> 
+  </data>
```
Comment (chlowell) [inline src/System.ComponentModel.TypeConverter/src/Resources/Strings.resx]: These are out of sorted order.

Comment (twsouthwick) [inline src/System.ComponentModel.TypeConverter/src/Resources/Strings.resx]: Fixed

**Comment by chlowell** [comment]
LGTM

**Comment by twsouthwick** [comment]
@dotnet-bot test please

**Pull request merged**